lib/Agrammon/DB/Tag.pm6

```diff
@@ -1,22 +1,59 @@
 use v6;
+use Agrammon::DB;
 use Agrammon::DB::User;
 
-class Agrammon::DB::Tag {
+class Agrammon::DB::Tag does Agrammon::DB {
     has Str $.name;
     has Int $.id;
     has Agrammon::DB::User $.user;
 
     method create {
         self.with-db: -> $db {
-            my $results = $db.query(q:to/DATASET/, $!name);
-                INSERT INTO tag (tag_name)
-                VALUES ($1)
+            my $results = $db.query(q:to/TAG/, $!name, $!user.id);
+                INSERT INTO tag (tag_name, tag_pers)
+                VALUES ($1, $2)
                 RETURNING tag_id
-            DATASET
+            TAG
 
             $!id = $results.value;
         }
         return self;
     }
 
+    method rename($new) {
+        self.with-db: -> $db {
+            my $ret = $db.query(q:to/TAG/, $new, $!name, $!user.id);
+                UPDATE tag SET tag_name = $1
+                 WHERE tag_name = $2 AND tag_pers = $3
+                RETURNING tag_name
+            TAG
+            $!name = $new;
+        }
+        return self;
+    }
+
+   method delete {
+        self.with-db: -> $db {
+            my $ret = $db.query(q:to/TAG/, $!id);
```

Comment:
Unused `$ret`

Reply:
Removed for the moment. I guess when checking the frontend interaction, I might actually want to return the number of tags deleted or something like that.